examples/webgl_gpgpu_water.html

```diff
@@ -550,6 +657,30 @@
 					uniforms.mousePos.value.set( 10000, 10000 );
 				}
 
+				// Read height value once in a time, since it has some cost
+				if ( ++numFrames > 120 ) {
+				
+					numFrames = 0;
+				
+					heightCompensation = - readWaterLevel();
+
+				}
+
+				// Apply gradually height compensation to reset water level
+				if ( heightCompensation !== 0 ) {
+				
+					var decremHeight = Math.min( 0.2, Math.abs( heightCompensation ) ) * Math.sign( heightCompensation );
```

Comment:
What is the reason for 0.2 ?

Reply:
It is just the fixed amount of drain every frame

Reply:
The exact amount is computed (`heightCompensation`), but applied progressively in each frame (`decremHeight`) after the reading so there is no visual bump in the water level.

You can test by editing `var decremHeight = heightCompensation; heightCompensation = 0;` and will see the visual bump each 2 seconds (or longer period if you want)

Reply:
As I said, why not decrement by height / 1200 progressively every frame for 1200 frames, then recompute hight and repeat.

Why does your formula contain `Math.min( 0.2, ... )` ?

Reply:
Now I understand, it makes sense. I've done a commit.

Reply:
I am not going to be able to invest any more time here...

Perhaps you will agree this is better.

```javascript
if ( ++ numFrames > 120 ) {

	numFrames = 0;

	// Apply gradually height compensation to reset water level

	heightmapVariable.material.uniforms.heightCompensation.value = - readWaterLevel() / 120;

}
```

Reply:
I'm sorry, but that would apply only one every 120 frames so it wouldn't be gradual. Also, the drain amount would be 1/120 times the needed one.